### PR TITLE
[GUILD-2561] - Email allowlist

### PR DESCRIPTION
**Pull request by cs-balazs**
- `RequirementAccessIndicator`: Turned out that a missing email is also returned as a platform error, so I added a condition for that as well. Started to get ugly, so added a const for the messages
- Added a `type` selector for the allowlist form with options `EVM and Fuel addresses` and `Email addresses`
- Validation uses `z.string().email()` just to be sure that it validates using the same email rules, as the schema used on BE
- Companion PRs:
  - Gate: https://github.com/guildxyz/guild-gate/pull/715
  - Core: https://github.com/guildxyz/guild-backend/pull/1369

### CODE REVIEW

**Review by dovalid — APPROVED** [review]

File: src/requirements/Allowlist/AllowlistForm.tsx [hunk 1]
```diff
@@ -43,16 +82,40 @@ const AllowlistForm = ({ baseFieldPath }: RequirementFormProps): JSX.Element =>
         ?.toString()
         ?.split("\n")
         ?.filter((line) => !!line)
-        ?.map((line) => line.slice(0, 42))
+        ?.map((line) =>
+          type === "ALLOWLIST_EMAIL" ? line.trim() : line.slice(0, 42)
```
Comment (BrickheadJohnny) [inline src/requirements/Allowlist/AllowlistForm.tsx:86]: I think this will "break" Fuel addresses (those are 66 long incl. 0x afaik) :ohno:

Comment (cs-balazs) [reply src/requirements/Allowlist/AllowlistForm.tsx:86]: I'm not sure if it currently works on prod then :D Should we just .trim always?

Comment (BrickheadJohnny) [reply src/requirements/Allowlist/AllowlistForm.tsx:86]: Oh 😂  Yeah, then let's just trim them 👍

Comment (cs-balazs) [reply src/requirements/Allowlist/AllowlistForm.tsx]: Made the change in https://github.com/guildxyz/guild.xyz/pull/1135/commits/7543acfad1b80896fb5735a0d7b6d9a7851031b2 👍

**Pull request merged**